emacs_anywhere.el

```diff
@@ -18,26 +19,42 @@
    (point-max)))
 
 (defun ea--gnu-linux-copy-to-clip ()
-  (write-region nil nil "/tmp/eaclipboard")
-  (shell-command "xclip -selection clipboard /tmp/eaclipboard &> /dev/null"))
+  (write-region nil nil "/tmp/eaclipboard"))
 
 (defun ea--delete-frame-handler (_frame)
+  (remove-hook 'delete-frame-functions 'ea--delete-frame-handler)
   (when ea-on
     (cond
      (ea--osx (ea--osx-copy-to-clip))
      (ea--gnu-linux (ea--gnu-linux-copy-to-clip)))
     (kill-buffer ea--buffer-name))
   (shell-command
-   (format "echo export EA_ABORT=%s > /tmp/eaenv"
-           (if ea-on "false" "true"))))
+   (format "echo export EA_ABORT=%s\";\" export EA_SHOULD_PASTE=%s > /tmp/eaenv"
+           (if ea-on "false" "true")
+           (if ea-paste "true" "false"))))
 
 (defun ea--init ()
   (setq ea-on t) ; begin each session with EA enabled
+  (setq ea-paste t) ; begin each session with paste enabled
   (add-hook 'delete-frame-functions 'ea--delete-frame-handler)
   (switch-to-buffer ea--buffer-name)
   (select-frame-set-input-focus (selected-frame))
   ;; run hook if defined
   (when (boundp 'ea-popup-hook)
```

Comment:
You can actually `defvar` the hook  in this script and provide a docstring and all.  In Emacs if someone creates a defvar after it was already created it will not overwrite the values, or if they setq it before it is defined it will also assume the old value.  This was partially done that way exactly because of hooks.

Reply:
Thanks @Fuco1 ! That was confusing. I'll fix this up.